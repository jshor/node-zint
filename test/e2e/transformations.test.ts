import OutputType from '../../src/types/enums/OutputType'
import OutputOption from '../../src/types/enums/OutputOption'
import SymbologyType from '../../src/types/enums/SymbologyType'
import { createImageFile } from '../helpers'
import symbology from '../../src'

describe('Symbology Transformations', () => {
  describe('rotations', () => {
    it('should rotate the symbol 90 degrees clockwise', async () => {
      const image = await createImageFile({
        symbology: SymbologyType.CODE128,
        rotation: 90
      }, OutputType.PNG, '12345')

      expect(image).toMatchImageSnapshot()
    })

    it('should rotate the symbol 180 degrees clockwise', async () => {
      const image = await createImageFile({
        symbology: SymbologyType.CODE128,
        rotation: 180
      }, OutputType.PNG, '12345')

      expect(image).toMatchImageSnapshot()
    })

    it('should rotate the symbol 270 degrees clockwise', async () => {
      const image = await createImageFile({
        symbology: SymbologyType.CODE128,
        rotation: 270
      }, OutputType.PNG, '12345')

      expect(image).toMatchImageSnapshot()
    })
  })

  describe('dot size', () => {
    it('should set the default dot size to 0.8px in dotty mode', async () => {
      const image = await createImageFile({
        symbology: SymbologyType.DOTCODE,
        outputOptions: OutputOption.BARCODE_DOTTY_MODE
      }, OutputType.PNG, '12345')

      expect(image).toMatchImageSnapshot()
    })

    it('should render the dot size as 1px in dotty mode', async () => {
      const image = await createImageFile({
        symbology: SymbologyType.DOTCODE,
        outputOptions: OutputOption.BARCODE_DOTTY_MODE,
        dotSize: 1
      }, OutputType.PNG, '12345')

      expect(image).toMatchImageSnapshot()
    })
  })

  describe('Scalable Vector Graphics', () => {
    it('should stream an SVG image', async () => {
      const image = await symbology.createStream({
        symbology: SymbologyType.CODE128,
      }, '12345', OutputType.SVG)

      expect(image.data).toMatchSnapshot()
    })

    it('should render an SVG file', async () => {
      const image = await createImageFile({
        symbology: SymbologyType.CODE128,
      }, OutputType.SVG, '12345')

      expect(image.toString()).toMatchSnapshot()
    })
  })

  describe('PostScript', () => {
    it('should stream a PostScript image', async () => {
      const image = await symbology.createStream({
        symbology: SymbologyType.CODE128,
      }, '12345', OutputType.EPS)

      expect(image.data).toMatchSnapshot()
    })

    it('should render an eps file', async () => {
      const image = await createImageFile({
        symbology: SymbologyType.CODE128,
      }, OutputType.EPS, '12345')

      expect(image.toString()).toMatchSnapshot()
    })
  })

  describe('Portable Network Graphics', () => {
    it('should stream a base64-encoded image', async () => {
      const image = await symbology.createStream({
        symbology: SymbologyType.CODE128
      }, '12345', OutputType.PNG)

      expect(image.data).toMatchSnapshot()
    })
  })
})
