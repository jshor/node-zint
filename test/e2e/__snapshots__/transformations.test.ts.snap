// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Symbology Transformations Portable Network Graphics should stream a base64-encoded image 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAJ4AAAB0CAYAAAB0WoufAAAAAklEQVR4AewaftIAAAJVSURBVO3BwW5cRxAEwazG/v8vlzkHwiP6iaBgQH3YjAhQLm05knC05ZaEJ225JeFoy5GEW1tuSXjSllsSjrY8ScLRliMJt7Y8ScLRliMJ32nLkYQnbTmScGvLkYQnbbkl4WjLLQlP2nIk4WjLkYRbW54k4daWJ0l40pYjCUdbjiTcBmnBIC0YpAWDtGCQFgzSgkFaMEgLBmnBIC0YpAWDtGCQFgzSgkFaMEgLBmnBIC0YpAWDtGCQFgzSgkFaMEgLBmnBIC0YpAWDtGCQFgzSgkFaMEgLBmnBIC0YpAWDtGCQFgzSgkFaMEgLBmnBIC0YpAWDtGCQFgzSgkFaMEgLBmnBIC0YpAWDtGCQFgzSgkFaMEgLBmnBIC0YpAWDtGCQFgzSgkFaMEgLBmnBIC0YpAWDtGCQFgzSgkFaMEgLBmnBIC0YpAWDtGCQFgzSgkFaMEgLBmnBIC0YpAWDtGCQFgzSgkFaMEgLBmnBIC0YpAWDtGCQFgzSgkFaMEgLBmnBIC0YpAXpB6S/bJAWDNKCQVrw4s0loS23JNzaciThSVu+SkJbjiQ8acu7evHGkvBVEtpyS0Jb2vJVEr5Kwldt0b+GN5WEtvwfSWjLLQlt0fdevKm2PGnLTyShLT+VhFtb3tkLfSsJbbkl4UhCWz4loS1P2nJLQlve1Qv9sbZ8SkJbvtMW/WrQbyWhLT+RhLboZwY9SkJbbkn4ThKSkIQjCUcS9KsX+o8ktOVPtOWWhLboWfqBN5SEr9qShCdtOZJwa8uTJLTlUxJubXln6Qekv2yQFvwDQxq16q7zH8sAAAAASUVORK5CYII="`;

exports[`Symbology Transformations PostScript should render an eps file 1`] = `
"%!PS-Adobe-3.0 EPSF-3.0
%%Creator: Zint 2.9.1.9
%%Title: Zint Generated Symbol
%%Pages: 0
%%BoundingBox: 0 0 158 119
%%EndComments
/TL { setlinewidth moveto lineto stroke } bind def
/TD { newpath 0 360 arc fill } bind def
/TH { 0 setlinewidth moveto lineto lineto lineto lineto lineto closepath fill } bind def
/TB { 2 copy } bind def
/TR { newpath 4 1 roll exch moveto 1 index 0 rlineto 0 exch rlineto neg 0 rlineto closepath fill } bind def
/TE { pop pop } bind def
newpath
1.00 1.00 1.00 setrgbcolor
118.90 0.00 TB 0.00 158.00 TR
TE
0.00 0.00 0.00 setrgbcolor
100.00 18.90 TB 0.00 4.00 TR
TE
100.00 18.90 TB 6.00 2.00 TR
TE
100.00 18.90 TB 12.00 6.00 TR
TE
100.00 18.90 TB 22.00 2.00 TR
TE
100.00 18.90 TB 26.00 4.00 TR
TE
100.00 18.90 TB 34.00 6.00 TR
TE
100.00 18.90 TB 44.00 2.00 TR
TE
100.00 18.90 TB 52.00 2.00 TR
TE
100.00 18.90 TB 56.00 4.00 TR
TE
100.00 18.90 TB 66.00 2.00 TR
TE
100.00 18.90 TB 70.00 8.00 TR
TE
100.00 18.90 TB 80.00 6.00 TR
TE
100.00 18.90 TB 88.00 4.00 TR
TE
100.00 18.90 TB 94.00 6.00 TR
TE
100.00 18.90 TB 104.00 2.00 TR
TE
100.00 18.90 TB 110.00 6.00 TR
TE
100.00 18.90 TB 118.00 2.00 TR
TE
100.00 18.90 TB 122.00 4.00 TR
TE
100.00 18.90 TB 132.00 4.00 TR
TE
100.00 18.90 TB 142.00 6.00 TR
TE
100.00 18.90 TB 150.00 2.00 TR
TE
100.00 18.90 TB 154.00 4.00 TR
TE
matrix currentmatrix
/Helvetica findfont
14.00 scalefont setfont
 0 0 moveto 79.00 3.50 translate 0.00 rotate 0 0 moveto
 (12345) stringwidth
pop
-2 div 0 rmoveto
 (12345) show
setmatrix
"
`;

exports[`Symbology Transformations PostScript should stream a PostScript image 1`] = `
"%!PS-Adobe-3.0 EPSF-3.0
%%Creator: Zint 2.9.1.9
%%Title: Zint Generated Symbol
%%Pages: 0
%%BoundingBox: 0 0 158 119
%%EndComments
/TL { setlinewidth moveto lineto stroke } bind def
/TD { newpath 0 360 arc fill } bind def
/TH { 0 setlinewidth moveto lineto lineto lineto lineto lineto closepath fill } bind def
/TB { 2 copy } bind def
/TR { newpath 4 1 roll exch moveto 1 index 0 rlineto 0 exch rlineto neg 0 rlineto closepath fill } bind def
/TE { pop pop } bind def
newpath
1.00 1.00 1.00 setrgbcolor
118.90 0.00 TB 0.00 158.00 TR
TE
0.00 0.00 0.00 setrgbcolor
100.00 18.90 TB 0.00 4.00 TR
TE
100.00 18.90 TB 6.00 2.00 TR
TE
100.00 18.90 TB 12.00 6.00 TR
TE
100.00 18.90 TB 22.00 2.00 TR
TE
100.00 18.90 TB 26.00 4.00 TR
TE
100.00 18.90 TB 34.00 6.00 TR
TE
100.00 18.90 TB 44.00 2.00 TR
TE
100.00 18.90 TB 52.00 2.00 TR
TE
100.00 18.90 TB 56.00 4.00 TR
TE
100.00 18.90 TB 66.00 2.00 TR
TE
100.00 18.90 TB 70.00 8.00 TR
TE
100.00 18.90 TB 80.00 6.00 TR
TE
100.00 18.90 TB 88.00 4.00 TR
TE
100.00 18.90 TB 94.00 6.00 TR
TE
100.00 18.90 TB 104.00 2.00 TR
TE
100.00 18.90 TB 110.00 6.00 TR
TE
100.00 18.90 TB 118.00 2.00 TR
TE
100.00 18.90 TB 122.00 4.00 TR
TE
100.00 18.90 TB 132.00 4.00 TR
TE
100.00 18.90 TB 142.00 6.00 TR
TE
100.00 18.90 TB 150.00 2.00 TR
TE
100.00 18.90 TB 154.00 4.00 TR
TE
matrix currentmatrix
/Helvetica findfont
14.00 scalefont setfont
 0 0 moveto 79.00 3.50 translate 0.00 rotate 0 0 moveto
 (12345) stringwidth
pop
-2 div 0 rmoveto
 (12345) show
setmatrix
"
`;

exports[`Symbology Transformations Scalable Vector Graphics should render an SVG file 1`] = `
"<?xml version="1.0" standalone="no"?>
<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN"
   "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">
<svg width="158" height="119" version="1.1"
   xmlns="http://www.w3.org/2000/svg">
   <desc>Zint Generated Symbol
   </desc>

   <g id="barcode" fill="#000000">
      <rect x="0" y="0" width="158" height="119" fill="#FFFFFF" />
      <rect x="0.00" y="0.00" width="4.00" height="100.00" />
      <rect x="6.00" y="0.00" width="2.00" height="100.00" />
      <rect x="12.00" y="0.00" width="6.00" height="100.00" />
      <rect x="22.00" y="0.00" width="2.00" height="100.00" />
      <rect x="26.00" y="0.00" width="4.00" height="100.00" />
      <rect x="34.00" y="0.00" width="6.00" height="100.00" />
      <rect x="44.00" y="0.00" width="2.00" height="100.00" />
      <rect x="52.00" y="0.00" width="2.00" height="100.00" />
      <rect x="56.00" y="0.00" width="4.00" height="100.00" />
      <rect x="66.00" y="0.00" width="2.00" height="100.00" />
      <rect x="70.00" y="0.00" width="8.00" height="100.00" />
      <rect x="80.00" y="0.00" width="6.00" height="100.00" />
      <rect x="88.00" y="0.00" width="4.00" height="100.00" />
      <rect x="94.00" y="0.00" width="6.00" height="100.00" />
      <rect x="104.00" y="0.00" width="2.00" height="100.00" />
      <rect x="110.00" y="0.00" width="6.00" height="100.00" />
      <rect x="118.00" y="0.00" width="2.00" height="100.00" />
      <rect x="122.00" y="0.00" width="4.00" height="100.00" />
      <rect x="132.00" y="0.00" width="4.00" height="100.00" />
      <rect x="142.00" y="0.00" width="6.00" height="100.00" />
      <rect x="150.00" y="0.00" width="2.00" height="100.00" />
      <rect x="154.00" y="0.00" width="4.00" height="100.00" />
      <text x="79.00" y="115.40" text-anchor="middle"
         font-family="Helvetica, sans-serif" font-size="14.0" >
         12345
      </text>
   </g>
</svg>
"
`;

exports[`Symbology Transformations Scalable Vector Graphics should stream an SVG image 1`] = `
"<?xml version="1.0" standalone="no"?>
<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN"
   "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">
<svg width="158" height="119" version="1.1"
   xmlns="http://www.w3.org/2000/svg">
   <desc>Zint Generated Symbol
   </desc>

   <g id="barcode" fill="#000000">
      <rect x="0" y="0" width="158" height="119" fill="#FFFFFF" />
      <rect x="0.00" y="0.00" width="4.00" height="100.00" />
      <rect x="6.00" y="0.00" width="2.00" height="100.00" />
      <rect x="12.00" y="0.00" width="6.00" height="100.00" />
      <rect x="22.00" y="0.00" width="2.00" height="100.00" />
      <rect x="26.00" y="0.00" width="4.00" height="100.00" />
      <rect x="34.00" y="0.00" width="6.00" height="100.00" />
      <rect x="44.00" y="0.00" width="2.00" height="100.00" />
      <rect x="52.00" y="0.00" width="2.00" height="100.00" />
      <rect x="56.00" y="0.00" width="4.00" height="100.00" />
      <rect x="66.00" y="0.00" width="2.00" height="100.00" />
      <rect x="70.00" y="0.00" width="8.00" height="100.00" />
      <rect x="80.00" y="0.00" width="6.00" height="100.00" />
      <rect x="88.00" y="0.00" width="4.00" height="100.00" />
      <rect x="94.00" y="0.00" width="6.00" height="100.00" />
      <rect x="104.00" y="0.00" width="2.00" height="100.00" />
      <rect x="110.00" y="0.00" width="6.00" height="100.00" />
      <rect x="118.00" y="0.00" width="2.00" height="100.00" />
      <rect x="122.00" y="0.00" width="4.00" height="100.00" />
      <rect x="132.00" y="0.00" width="4.00" height="100.00" />
      <rect x="142.00" y="0.00" width="6.00" height="100.00" />
      <rect x="150.00" y="0.00" width="2.00" height="100.00" />
      <rect x="154.00" y="0.00" width="4.00" height="100.00" />
      <text x="79.00" y="115.40" text-anchor="middle"
         font-family="Helvetica, sans-serif" font-size="14.0" >
         12345
      </text>
   </g>
</svg>
"
`;
